.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,68 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+# ******** NOTE ********
+
+name: "CodeQL"
+
+on:
+  push:
```

Comment:
Running on push is very risky as ppl would commit several times a day but probably wont open a PR.

Reply:
You can use this: https://github.com/opendistro-for-elasticsearch/index-management/blob/master/.github/workflows/test-and-build-workflow.yml#L3-L11
Modify as appropriate

Reply:
Removed the push

Reply:
Added new pull and push configuration